### PR TITLE
[Backport 8.4][DOCS] Removes outdated pages from docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR backports the changes of https://github.com/elastic/elasticsearch-net/pull/6655 to the 8.4 branch.

### CODE REVIEW

File: docs/redirects.asciidoc [hunk 1]
```diff
@@ -6,4 +6,19 @@ The following pages have moved or been deleted.
...
+== NEST - High level client
+
+This page has been deleted.
+
+[role="exclude",id="indexing-documents"]
+== Indexing documents
+
+This page has been deleted.
+
+[role="exclude",id="bulkall-observable"]
```
Comment (szabosteve) [inline docs/redirects.asciidoc:21]:
```suggestion
[role="exclude",id="indexing-documents.html#bulkall-observable"]
```

File: docs/redirects.asciidoc [hunk 2]
```diff
@@ -6,4 +6,19 @@ The following pages have moved or been deleted.
...
+== NEST - High level client
+
+This page has been deleted.
+
+[role="exclude",id="indexing-documents"]
+== Indexing documents
+
+This page has been deleted.
+
+[role="exclude",id="indexing-documents.html#bulkall-observable"]
```
Comment (szabosteve) [inline docs/redirects.asciidoc:21]:
```suggestion
[role="exclude",id="#bulkall-observable"]
```

File: docs/redirects.asciidoc [hunk 3]
```diff
@@ -6,4 +6,19 @@ The following pages have moved or been deleted.
...
+== NEST - High level client
+
+This page has been deleted.
+
+[role="exclude",id="indexing-documents"]
+== Indexing documents
+
+This page has been deleted.
+
+[role="exclude",id="#bulkall-observable"]
```
Comment (szabosteve) [inline docs/redirects.asciidoc:21]:
```suggestion
[role="exclude",id="bulkall-observable"]
```

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Review by stevejgordon — APPROVED** [review]
LGTM!

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Comment by szabosteve** [comment]
@elasticmachine test this please

**Pull request merged**